mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/model/MaildirMessage.java

```diff
@@ -274,17 +275,22 @@ public InputStream getHeaderContent() throws IOException {
     @Override
     public List<MessageAttachment> getAttachments() {
         try {
+            AtomicInteger counter = new AtomicInteger(0);
             return new MessageParser().retrieveAttachments(getFullContent())
                 .stream()
                 .map(Throwing.<ParsedAttachment, MessageAttachment>function(
-                    attachmentMetadata -> attachmentMetadata.asMessageAttachment(AttachmentId.random()))
+                    attachmentMetadata -> attachmentMetadata.asMessageAttachment(generateFixedAttachmentId(counter.incrementAndGet())))
                     .sneakyThrow())
                 .collect(Guavate.toImmutableList());
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    private AttachmentId generateFixedAttachmentId(int position) {
+        return AttachmentId.from(messageName.getFullName() + "-" + position);
```

Comment:
I still don't understand why we need this. Can we have a global view on the design before/after, it's really hard to me to figure how it improves things

Reply:
Because the results needs to be deterministic otherwise it breaks some tests (from what I remind, sorry, when I implemented this fix, it was MOUNTHS agao, I don't recall well.)